### PR TITLE
Ben/add longformer

### PR DESCRIPTION
**Pull request by benleetownsend**

### CODE REVIEW

File: finetune/util/huggingface_interface.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def preprocess_for_alignment(text):
 
 
 def load_weights_from_hdf5_group_by_name(filepath, weights_replacement):
+    print(filepath)
```
Comment (madisonmay) [inline finetune/util/huggingface_interface.py]: Rogue print

File: finetune/util/huggingface_interface.py [hunk 2]
```diff
@@ -206,6 +207,9 @@ def _encode(self, texts):
                     token_ends = []
                     token_starts = []
                     for start, end in encoded.offsets:
+                        if token_ends:
+                            start = max(token_ends[-1], start)
```
Comment (madisonmay) [inline finetune/util/huggingface_interface.py:237]: Super bizarre.  Any idea why this occurs?

**Comment by madisonmay** [comment]
@benleetownsend do you think it's worth adding an activation check, or overrated?

File: finetune/util/huggingface_interface.py (hunk 2 above)
Comment (benleetownsend) [reply finetune/util/huggingface_interface.py:237]: This is really just a difference between how huggingface and we think about chars that become more than 1 token.
They repeat the start and end, but we assign it to one of the tokens and set the other to 0 length.

**Pull request merged**